### PR TITLE
TST: split up tests/plotting/test_frame.py into subdir & modules #34769

### LINKED ISSUE #34769: TST: split up tests/plotting/test_frame.py into subdir & modules
we should split up this test file into a separate subdir and multiple test modules for better groking. first PR should be a straight move, then can split (or all in one is ok)

### PR DESCRIPTION
**Pull request by Mikhaylov-yv**
Created instead of #37538

- [x] closes #34769
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

I moved the file to a new directory.
What is the best structure for this folder?

[before_test_ploting_frame](https://gist.github.com/Mikhaylov-yv/999de1451e8ecc836215b30db34ef841)
[after_test_ploting_frame](https://gist.github.com/Mikhaylov-yv/98eca32c21f59cc9e7653cd96d5fabd2)

### CODE REVIEW

**Comment by jreback** [comment]
in pre-commit:

```pandas/tests/plotting/frame/test_frame_groupby.py:7:1: F401 'pandas as pd' imported but unused```

also merge master and ping on green.

**Comment by Mikhaylov-yv** [comment]
I am very sorry. I don’t understand what to do next?

**Comment by jbrockmendel** [comment]
> I am very sorry. I don’t understand what to do next?

run `git pull upstream master` to merge master, then in the new files use `from pandas import foo` instead of `pd.foo`

**Comment by Mikhaylov-yv** [comment]
Thanks @jbrockmendel. After that, I get a message like this:
image
Am I doing the right thing.

**Comment by jbrockmendel** [comment]
> Am I doing the right thing?

I dont see anything wrong in the screenshot you posted.

Near the bottom of this page you should see a box with red text saying "Some checks were not successful".  In that box there are two red "x"s for the failed checks.  Click through to the details on those to see what still needs to be addressed.

File: pandas/tests/plotting/frame/test_frame_color.py [hunk 1]
```diff
@@ -0,0 +1,632 @@
...
+        ax = df.loc[:, [0]].plot.bar(color="DodgerBlue")
+        self._check_colors([ax.patches[0]], facecolors=["DodgerBlue"])
+        tm.close()
+
+        ax = df.plot(kind="bar", color="green")
+        self._check_colors(ax.patches[::5], facecolors=["green"] * 5)
+        tm.close()
+
+    def test_bar_user_colors(self):
+        df = pd.DataFrame(
```
Comment (jbrockmendel) [inline pandas/tests/plotting/frame/test_frame_color.py:158]: the linter will want this to be just DataFrame

**Comment by Mikhaylov-yv** [comment]
@jbrockmendel Thanks. I think I figured out what to do.

**Comment by Mikhaylov-yv** [comment]
Can you tell me what to do with this error message? This import is used in the **pandas / _testing.py** file

> Check for non-standard numpy.random-related imports excluding pandas/_testing.py.......................Failed
> - hook id: non-standard-numpy.random-related-imports
> - exit code: 1
> 
> pandas/tests/plotting/frame/test_frame.py:9:from numpy.random import rand, randn
> pandas/tests/plotting/frame/test_frame_color.py:6:from numpy.random import rand, randn

Do I need to fix **rand()** to the **np.random.rand()**

**Comment by Mikhaylov-yv** [comment]
I hope I took into account all my mistakes

**Comment by Mikhaylov-yv** [comment]
The errors don't seem to be relevant to my pull request. I'm wrong?

**Comment by jbrockmendel** [comment]
> The errors don't seem to be relevant to my pull request. I'm wrong?

You are not wrong.  This is affecting all the builds, you can ignore this failure for now.

**Comment by Mikhaylov-yv** [comment]
Thanks @jbrockmendel Does this mean the job is ready?  or do you need to do something else?

**Comment by jbrockmendel** [comment]
@Mikhaylov-yv is this pretty much cut/paste?  if so, can you merge master and ping on green

**Comment by Mikhaylov-yv** [comment]
@jbrockmendel Yes it is.  the number of tests is the same as the master.  
Now I'm going to merge

**Comment by jbrockmendel** [comment]
LGTM cc @jreback

**Comment by jreback** [comment]
thanks @Mikhaylov-yv

**Comment by Mikhaylov-yv** [comment]
@jreback @jbrockmendel  Thank you very much for your help

**Pull request merged**